### ISSUE TITLE
Read error: #<OpenURI::HTTPError: 403 Forbidden>

### ISSUE THREAD

**Issue by gusMoreno** [fringd/zipline#46]
Hi,

I followed the docs, and I am receiving the current error:
Read error: #<OpenURI::HTTPError: 403 Forbidden>

My files are encrypted and private in S3, so the URL i am passing to file_mappings is a secure_presigned_url that is returned from this method `S3Helper.new().secure_presigned_url(file.s3_key)`

The `file_mappings` variable is created as an array as per zipline docs.

At first I thought I wasn't generating the secure urls properly, but I pasted the generated urls directly in the browser, and the download works fine, so I don't understand why it is throwing a forbidden.

Could you please help me?

Bellow is my code, and console response.

```ruby
  def download_all_filtered
    require 'open-uri'
    files = Arquivo.where("id IN (#{params[:file_ids]})")
    files_for_zipline = files.map{ |file| [S3Helper.new().secure_presigned_url(file.s3_key), file.nome]}
    file_mappings = files_for_zipline.lazy.map { |url, path| [open(url), path] }
    zipline(file_mappings, 'teste.zip')
  end
```

```
Read error: #<OpenURI::HTTPError: 403 Forbidden>
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:377:in `open_http'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:755:in `buffer_open'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:226:in `block in open_loop'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:224:in `catch'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:224:in `open_loop'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:165:in `open_uri'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:735:in `open'
/Users/ownee/.rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/open-uri.rb:35:in `open'
/Users/ownee/ownee_web/app/controllers/pins_controller.rb:173:in `block in download_all_filtered'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:18:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:18:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:18:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:18:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:18:in `block in each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zip_tricks-4.7.1/lib/zip_tricks/streamer.rb:106:in `open'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/zipline-1.0.2/lib/zipline/zip_generator.rb:17:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/actionpack-5.1.6.1/lib/action_dispatch/http/response.rb:145:in `each_chunk'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/actionpack-5.1.6.1/lib/action_dispatch/http/response.rb:126:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/actionpack-5.1.6.1/lib/action_dispatch/http/response.rb:74:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/actionpack-5.1.6.1/lib/action_dispatch/http/response.rb:473:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/rack-2.0.6/lib/rack/body_proxy.rb:36:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/rack-2.0.6/lib/rack/body_proxy.rb:36:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/rack-2.0.6/lib/rack/body_proxy.rb:36:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/rack-2.0.6/lib/rack/body_proxy.rb:36:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/rack-2.0.6/lib/rack/body_proxy.rb:36:in `each'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/puma-3.12.0/lib/puma/server.rb:791:in `handle_request'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/puma-3.12.0/lib/puma/server.rb:472:in `process_client'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/puma-3.12.0/lib/puma/server.rb:332:in `block in run'
/Users/ownee/.rvm/gems/ruby-2.5.0/gems/puma-3.12.0/lib/puma/thread_pool.rb:133:in `block in spawn_thread'
```
Thanks in advance!

**Comment by fringd** [comment]
I am not sure, but it sounds like an s3 problem

**Comment by fringd** [comment]
Does the url you're using in the browser work in an incognito window?

**Comment by gusMoreno** [comment]
Yes, just tested it and it works in an incognito window.

**Comment by gusMoreno** [comment]
In fact, I continued testing, and there was an error with one of the files in the list. The download of this file does not work even by browser url. So I removed the file with problems and the others are working fine via browser url. 

The weird thing now is that noting happens. The server throws a `completed 200 OK` message, but I don't see the download anywhere. Did I miss anything in the method I posted above?
